configure

```diff
@@ -2,6 +2,7 @@
 
 prefix=/usr/local
 OUR_CPPFLAGS=
+C99_CFLAGS="-std=c99"
```

Comment:
what's the point of separating this out ? i can't see it conditionally used later on

Reply:
There are 2 options not recognized by Solaris ld `--no-as-needed` and `-soname=<name>`

`--no-as-needed`  is not need for Solaris ld, since it is not have such bug to workaround by this option.

`-soname=<name>` could be replaced to `-h <name>` or `--soname <name>`, and the later 2 forms are also accepted by GNU ld.